daemon/src/engine/server/CryptoChallege.h

```diff
@@ -0,0 +1,633 @@
+/*
```

Comment:
Sure, AES256 is a good standard. As for error reporting, if you really want you can stick a Log::Logger's logger.Notice in there (note that it shouldn't be on by default as we should assume stuff to work)